### PR TITLE
[#2636] Abbreviation for Resource Profile 4.0

### LINKED ISSUE #2636: Add Abbreviation to Resource form (Resource Profile v4.0)
Add field Abbreviation to section Basic in a Resource form below Name
https://marketplace.docker-fid.grid.cyf-kr.edu.pl/backoffice/services/new

image

https://confluence.egi.eu/display/EOSCEN/EOSC+Resource+Profile+v4.00

Field should not be required in Marketplace

Abbreviations should be displayed near the name
image

### PR DESCRIPTION
**Pull request by goreck888**
Add abbreviation string input to the Service model
Closes #2636

### CODE REVIEW

File: app/views/backoffice/services/_form.html.haml [hunk 1]
```diff
@@ -51,9 +51,11 @@
                     %i.fas.fa-chevron-up
         .collapse{ id: "basic", "aria-labelledby": "basic-header",
             "data-parent": "#accordion-form",
-            class: ("show" unless collapsed?(service, [:name, :resource_organisation, :providers, :webpage_url])) }
+            class: ("show" unless collapsed?(service, [:name, :abbreviation, :resource_organisation,
```
Comment (kmarszalek) [inline app/views/backoffice/services/_form.html.haml:54]: missing :abbreviation at line 41?

File: app/components/presentable/header_component.rb [hunk 2]
```diff
@@ -17,11 +17,13 @@ def initialize(
     preview:,
     question:,
     favourite_services: [],
+    abbreviation: nil,
```
Comment (kmarszalek) [inline app/components/presentable/header_component.rb:20]: why nil?

File: app/components/presentable/header_component.html.haml [hunk 3]
```diff
@@ -8,6 +8,8 @@
         = presentable_logo(@object)
     .col-12.col-sm-9.service-details-header
       %h2.font-weight-bolder= @title
+      - if @abbreviation
```
Comment (kmarszalek) [inline app/components/presentable/header_component.html.haml:11]: it's mandatory..

**Review by kmarszalek — CHANGES_REQUESTED** [review]
get @michal-szostak to approve of the no-mandatory approach (that will be visible in the PR) or make adjustments

File: app/components/presentable/header_component.html.haml (hunk 3 above)
Comment (goreck888) [reply app/components/presentable/header_component.html.haml:11]: The same reason, even if `service.abbreviation` is mandatory, the `@abbreviation` variable could be `nil`

**Comment by Marcelinna** [comment]
Abbreviation should be not mandatory in Marketplace

**Review by kmarszalek — APPROVED** [review]

**Review by kmarszalek — APPROVED** [review]
Squash :)

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**